zerver/migrations/0281_add_and_set_message_has_reaction.py

```diff
@@ -0,0 +1,38 @@
+from django.db import migrations, models
+from django.db.backends.postgresql.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+def set_has_reaction_in_messages(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    Message = apps.get_model("zerver", "Message")
+    Reaction = apps.get_model("zerver", "Reaction")
+    message = Message.objects.all()
+
+    for e in message:
+        if not e.has_reaction and Reaction.objects.filter(message=e).exists():
+            e.has_reaction = True
+            e.save(update_fields=["has_reaction"])
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('zerver', '0280_userprofile_presence_enabled'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='archivedmessage',
+            name='has_reaction',
+            field=models.BooleanField(db_index=True, default=False),
+        ),
+        migrations.AddField(
+            model_name='message',
+            name='has_reaction',
+            field=models.BooleanField(db_index=True, default=False),
```

Comment:
The first time I tried to run it with db_index=True it gave the following error:

`django.db.utils.OperationalError: cannot CREATE INDEX "zerver_message" because it has pending trigger events`

But know it's working, does anyone know why that happened?